src/test/java/org/logstash/beats/BeatsParserTest.java

```diff
@@ -50,14 +53,25 @@ public void testEncodingDecodingFields() {
         assertMessages(this.batch, decodedBatch);
     }
 
-
     @Test
-    public void  testCompressedEncodingDecodingFields() {
+    public void testCompressedEncodingDecodingFields() {
         this.batch.setProtocol(Protocol.VERSION_1);
         Batch decodedBatch = decodeCompressedBatch();
         assertMessages(this.batch, decodedBatch);
     }
 
+    @Test
+    public void testShouldNotCrashOnGarbageData() {
```

Comment:
Just to clarify that, the exception **need** to be propagated down to the `BeatsHandler`, because the handler will notify the ruby listener so the connections can be correctly removed from the connection list and the buffered data flushed down the pipeline.